src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -1167,13 +1211,44 @@ public static void AddPackageDependency(PackageDependency dependency, ISet<Packa
                 }
                 else
                 {
-                    throw new PackagingException(NuGetLogCode.NU5016, string.Format(CultureInfo.CurrentCulture, Strings.Error_InvalidDependencyVersionConstraints, dependency.Id));
+                    throw new PackagingException(
+                        NuGetLogCode.NU5016,
+                        string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.Error_InvalidDependencyVersionConstraints,
+                            dependency.Id));
                 }
             }
             else
             {
                 set.Add(dependency);
             }
         }
+
+        private static Regex[] GetDefaultExcludes()
+        {
+            var options = RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture;
+
+            if (Path.DirectorySeparatorChar == '/')
+            {
+                return new[]
+                {
+                    // Exclude previous package files
+                    new Regex(@"\.nupkg$", options),
+                    // Exclude all files and directories that begin with "."
+                    new Regex(@"/\.[^/.]*$", options),
+                    new Regex(@"^\.[^/.]*$", options)
+                };
+            }
+
+            return new[]
+            {
+                // Exclude previous package files
+                new Regex(@"\.nupkg$", options),
+                // Exclude all files and directories that begin with "."
+                new Regex(@"\\\.[^\.]*$", options),
+                new Regex(@"^\.[^\.]*$", options)
+            };
+        }
```

Comment:
For comparison:

Windows:

Before | After
--- | ---
`^(.+\\)*[^\\]*(\\)?\.nupkg$` | `\.nupkg$`
`^(.+\\)*\\\.[^\\.]*$` | `\\\.[^\.]*$`
`^\.[^\\.]*$` | `^\.[^\.]*$`

Linux/macOS:

Before | After
--- | ---
`^(.+/)*[^/]*(/)?\.nupkg$` | `\.nupkg$`
`^(.+/)*/\.[^/.]*$` | `/\.[^/.]*$`
`^\.[^/.]*$` | `^\.[^/.]*$`

Reply:
I can't be the only one who finds regex's incomprehensible gibberish. Do these regex's have any performance benefit over:

```csharp
var filename = Path.GetFileName(fullpath);
if (filename.EndsWith(".nupkg") || filename.StartsWith("."))
{
  // do whatever
}
```

I know which I find more readable. It took me an unreasonable amount of time to understand the difference between `\\\.[^\.]*$` and `^\.[^\.]*$`

Reply:
Good feedback.  There isn't a significant performance difference one way or the other.  I tested your suggestion and the numbers were comparable:

Version | Average | 95th Percentile
-- | -- | --
5.5.x | 3.046 | 3.290

In another trial, the numbers were nearly identical to the original numbers.

There is a functional difference, however.  For what appears to be a bug in the original implementation (even in NuGet 3.5.0) the "exclude file names beginning with dot (.)" regular expression pattern did not exclude files beginning with a dot and containing another dot.  So, a file name like `.jscs.json` would not be excluded.  Whether or not that is a bug doesn't matter here; in this PR I want to avoid functional behavior changes except to the fix the regression in https://github.com/NuGet/Home/issues/8798.

That said, I've taken your excellent suggestion and modified it slightly to preserve the existing behavior.